src/main/java/uk/gov/hmcts/reform/pip/account/management/service/MediaApplicationService.java

```diff
@@ -0,0 +1,157 @@
+            new NotFoundException(String.format("Application with id %s could not be found", id)));
+    }
+
+    /**
+     * Get an image from the blob store by the image id.
+     *
+     * @param imageId The id of the image to retrieve
+     * @return The file if found
+     */
+    public Resource getImageById(String imageId) {
+        try {
+            return azureBlobService.getBlobFile(imageId);
+        } catch (BlobStorageException e) {
+            throw new NotFoundException(String.format("Image with id %s could not be found", imageId));
+        }
+    }
+
+    /**
+     * Create an application and store the image in the blob store, saving the blob url in the entity.
+     *
+     * @param application The application entity to save to the database
+     * @param file The file to upload to the blob store
+     * @return The newly created application
+     */
+    public MediaApplication createApplication(MediaApplication application, MultipartFile file) {
+        String imageId = azureBlobService.uploadFile(UUID.randomUUID().toString(), file);
+
+        application.setRequestDate(LocalDateTime.now());
+        application.setStatusDate(LocalDateTime.now());
+        application.setImage(imageId);
+        application.setImageName(file.getOriginalFilename());
+
+        return mediaApplicationRepository.save(application);
+    }
+
+    /**
+     * Update an application by fetching the application.
+     *
+     * @param id The id of the application to update
+     * @param status The status to update the application with
+     * @return The updated application
+     */
+    public MediaApplication updateApplication(UUID id, MediaApplicationStatus status) {
+        MediaApplication applicationToUpdate = mediaApplicationRepository.findById(id).orElseThrow(() ->
+            new NotFoundException(String.format("Application with id %s could not be found", id)));
+
+        applicationToUpdate.setStatus(status);
+        applicationToUpdate.setStatusDate(LocalDateTime.now());
+
+        return mediaApplicationRepository.save(applicationToUpdate);
+    }
+
+    /**
+     * Delete an application by fetching the application entity to ensure the correct blob will be deleted.
+     *
+     * @param id The id of the application to delete
+     */
+    public void deleteApplication(UUID id) {
+        MediaApplication applicationToDelete = mediaApplicationRepository.findById(id).orElseThrow(() ->
+            new NotFoundException(String.format("Application with id %s could not be found", id)));
+
+        azureBlobService.deleteBlob(applicationToDelete.getImage());
+        mediaApplicationRepository.delete(applicationToDelete);
+    }
+
+    /**
+     * Scheduled job that gets all media applications & sends them to publication services.
+     */
+    @Scheduled(cron = "${cron.media-application-reporting}")
+    public void processApplicationsForReporting() {
+        List<MediaApplication> mediaApplications = getApplications();
```

Comment:
I'm thinking rather than getting all applications from the database here. Could we filter the applications returned at the repository layer? This would mean we don't need to do any filtering in the subsequent methods for only APPROVED / REJECTED

Reply:
I think we need to get all applications here as we want all to be sent regardless of status, I could however have them fetched again in the delete and have it filtered at repository layer if that would be better?